### PR TITLE
Victor/adding wallet applications created date kin2 3

### PR DESCRIPTION
**Pull request by Doc999tor**
- [ ] linked to a JIRA issue
- [ ] description of the change
- [ ] tests for the newly added code
- [ ] if this is a breaking change, contains backward support code
- [ ] if this requires client changes or partner updates, updated dependendants
- [ ] were metrics and alerts added
- [ ] should I change documentation

### CODE REVIEW

File: docker-compose.deps.yaml [hunk 1]
```diff
@@ -80,6 +80,7 @@ services:
       OLD_HORIZON: 'https://horizon-playground.kininfrastructure.com'
       NEW_HORIZON: 'https://horizon-testnet.kininfrastructure.com'
       NEW_PASSPHRASE: 'Kin Testnet ; December 2018'
+      APP_INTERNAL_SERVICE: 'http://marketplace-internal'
```
Comment (liorama) [inline docker-compose.deps.yaml]: I think it's better to have it only in the migration server.

File: scripts/src/internal/routes.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
 	walletCreationSuccess as walletCreationSuccessService,
 	WalletCreationSuccessData,
 	WalletCreationFailureData, FailedPayment,
+	burnWallet as burnWalletService,
```
Comment (liorama) [inline scripts/src/internal/routes.ts]: Since the service doesn't burn the wallet maybe `markWalletBurnt`

File: scripts/src/internal/routes.ts [hunk 3]
```diff
@@ -83,11 +84,21 @@ export const getJwtKeys = async function(req: WebHookRequest, res: Response) {
 	res.status(200).send(PUBLIC_KEYS);
 } as any as RequestHandler;
 
+type BurnWalletRequest = Request & {
```
Comment (liorama) [inline scripts/src/internal/routes.ts]: `MarkWalletBurntRequest`

File: scripts/src/internal/routes.ts [hunk 4]
```diff
@@ -83,11 +84,21 @@ export const getJwtKeys = async function(req: WebHookRequest, res: Response) {
...
+	params: { wallet_address: string }
+};
+export const burnWallet = async function(req: BurnWalletRequest, res: Response) {
```
Comment (liorama) [inline scripts/src/internal/routes.ts]: Add a line break before

File: scripts/src/internal/routes.ts [hunk 5]
```diff
@@ -83,11 +84,21 @@ export const getJwtKeys = async function(req: WebHookRequest, res: Response) {
...
+	await burnWalletService(walletAddress);
+	res.sendStatus(204);
+} as any as RequestHandler;
+
 export function createRoutes(app: Express, pathPrefix?: string) {
 	const router = Router();
 	router
 		.post("/webhook", webhookHandler)
-		.get("/jwt-keys", getJwtKeys);
+		.get("/jwt-keys", getJwtKeys)
```
Comment (liorama) [inline scripts/src/internal/routes.ts]: Why remove the `;`?

File: scripts/src/internal/routes.ts [hunk 6]
```diff
@@ -83,11 +84,21 @@ export const getJwtKeys = async function(req: WebHookRequest, res: Response) {
...
+		.put("/wallets/:wallet_address/burn", burnWalletService);
```
Comment (liorama) [inline scripts/src/internal/routes.ts]: `"/wallets/:wallet_address/burnt"`

Comment (liorama) [reply scripts/src/internal/routes.ts]: Maybe it should  be `"/wallets/kin2/:wallet_address/burn"` for clarity

File: scripts/src/models/users.ts [hunk 7]
```diff
@@ -240,4 +240,10 @@ export class WalletApplication extends BaseEntity {
 
 	@Column({ name: "app_id" })
 	public appId!: string;
+
+	@Column({ name: "created_date_kin2", nullable: true })
+	public createdDateKin2?: Date;
+
+	@Column({ name: "created_date_kin3", nullable: true })
+	public createdDateKin3?: Date;
```
Comment (liorama) [inline scripts/src/models/users.ts]: Didn't we say Kin3 isn't  a good name?

**Review by liorama — CHANGES_REQUESTED** [review]

File: scripts/src/public/services/migration.ts [hunk 8]
```diff
@@ -27,18 +29,12 @@ export const accountStatus = async function(req: express.Request, res: express.R
 
 	const app_blockchain_version = await getAppBlockchainVersion(appId);
 
-	const isBurnedRes = await client.get(`${ getConfig().migration_service }/accounts/${ publicAddress }/status`, {
```
Comment (liorama) [inline scripts/src/public/services/migration.ts]: Remove the creation of `client`, axios, axios-retry and everything else that becomes irrelevant or unused after this edit.

File: scripts/src/public/services/migration.ts [hunk 9]
```diff
@@ -27,18 +29,12 @@ export const accountStatus = async function(req: express.Request, res: express.R
...
-		}
-	});
-	if (isBurnedRes.status === 404) {
-		isBurnedRes.data.is_burned = true; // Wallet doesn't exist
-	} else if (isBurnedRes.status > 400) {
-		logger().info(`Something went wrong, migration service response`, isBurnedRes);
-		throw Error(`Something went wrong, migration service response: ${ JSON.stringify(isBurnedRes.data) }`);
+	const wallet = await WalletApplication.findOne({ walletAddress: publicAddress });
+	if (!wallet) {
+		throw NoSuchWallet(publicAddress);
```
Comment (liorama) [inline scripts/src/public/services/migration.ts]: @doodyparizada WDTY, should this throw or if the `app_blockchain_version` is 3 and no wallet maybe return `shouldMigrate: true`.

File: scripts/src/internal/routes.ts (hunk 5 above)
Comment (Doc999tor) [reply scripts/src/internal/routes.ts]: It chained further to `.put` request

**Review by liorama — APPROVED** [review]

File: scripts/src/public/services/migration.ts (hunk 9 above)
Comment (doodyparizada) [reply scripts/src/public/services/migration.ts]: pretty sure it should behave as before when receiving `404`:
```javascript
if (isBurnedRes.status === 404) {
		isBurnedRes.data.is_burned = true; // Wallet doesn't exist
```

**Pull request merged**